tests/data_types/tensor.cpp

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: MIT
+#include <ddc/ddc.hpp>
+
+#include <gtest/gtest.h>
+
+#include "tensor_index_tools.hpp"
+#include "vector_index_tools.hpp"
+
+using namespace tensor_tools;
+
+namespace {
+struct R_cov;
+struct Theta_cov;
+
+struct R
+{
+    static bool constexpr PERIODIC = false;
+    static bool constexpr IS_COVARIANT = false;
+    static bool constexpr IS_CONTRAVARIANT = true;
+    using Dual = R_cov;
+};
+struct Theta
+{
+    static bool constexpr PERIODIC = true;
+    static bool constexpr IS_COVARIANT = false;
+    static bool constexpr IS_CONTRAVARIANT = true;
+    using Dual = Theta_cov;
+};
+
+struct R_cov
+{
+    static bool constexpr PERIODIC = false;
+    static bool constexpr IS_COVARIANT = true;
+    static bool constexpr IS_CONTRAVARIANT = false;
+    using Dual = R;
+};
+struct Theta_cov
+{
+    static bool constexpr PERIODIC = true;
+    static bool constexpr IS_COVARIANT = true;
+    static bool constexpr IS_CONTRAVARIANT = false;
+    using Dual = Theta;
+};
+
+TEST(TensorTools, TensorIndexSet)
+{
+    using IdxSet = VectorIndexSet<R, Theta>;
+    using IdxSet_cov = VectorIndexSet<R_cov, Theta_cov>;
+    using TestIndexSet = TensorIndexSet<IdxSet, IdxSet, IdxSet_cov>;
+    static_assert(TestIndexSet::rank() == 3);
+    static_assert(TestIndexSet::size() == 8);
+    static_assert(std::is_same_v<TestIndexSet::get_vector_index_set_along_dim_t<0>, IdxSet>);
+    static_assert(std::is_same_v<TestIndexSet::get_vector_index_set_along_dim_t<1>, IdxSet>);
+    static_assert(std::is_same_v<TestIndexSet::get_vector_index_set_along_dim_t<2>, IdxSet_cov>);
+    using TestIndexElement0 = get_nth_tensor_index_element_t<0, TestIndexSet>;
```

Comment:
Could a syntax like the following be preferable ?

```suggestion
    using TestIndexElement000 = get_tensor_index_element_t<Indices<0, 0, 0>, TestIndexSet>;
```

Reply:
Hmm, maybe my comments aren't clear enough. I think `get_nth_tensor_index_element_t` does the opposite of what you are suggesting. Unless you are suggesting to then save the tensor in a multi-D Kokkos object?

Reply:
We could think of using mdspan that supports compile-time dimensions. You wouldn't need to do the linearization by hand.